src/unitquaternion.jl

```diff
@@ -365,6 +365,43 @@ function kinematics(q::Q, ω::AbstractVector) where Q <: UnitQuaternion
     1//2 * params(q*Q(0.0, ω[1], ω[2], ω[3], false))
 end
 
+# ~~~~~~~~~~~~~~~ Spherical Linear Interpolation ~~~~~~~~~~~~~ #
+
+function slerp(qa::UnitQuaternion{T}, qb::UnitQuaternion{T}, t, ϵ=0.001) where T
```

Comment:
`ϵ` should depend on `T`
See `tol` in `isrotation`. (https://github.com/JuliaGeometry/Rotations.jl/blob/v1.0.4/src/core_types.jl#L213)

Reply:
should we have a type promotion for `qa` and `qb` so they don't need to be the same type?
something like:
```julia
function slerp(qa::UnitQuaternion, qb::UnitQuaternion, t::Real, eps::Union{Real, Nothing}=nothing)
    T = promote_type(eltype(qa), eltype(qb))
    eps = eps === nothing ? 1000 * _isrotation_eps(eltype(T)) : eps
    ...
end
```

```

Reply:
After spending a bit more time looking into this, I'm thinking we should let the result of interpolating between quaternions with angle difference of 180 degrees be arbitrary, rather than defining a default interpolation path. We can just alert the users of this in the docstring, and keep the slerp function simpler and faster. Thoughts?

Reply:
I think it's better to use multiple dispatches here.
```julia
function slerp(qa::UnitQuaternion{Ta}, qb::UnitQuaternion{Tb}, t) where {Ta, Tb}
    T = promote_type(Ta, Tb)
    return(UnitQuaternion{T}(qa), UnitQuaternion{T}(qb), t)
end

function slerp(qa::UnitQuaternion{T}, qb::UnitQuaternion{T}, t) where T
    ϵ = 1000 * _isrotation_eps(eltype(T))
    ...
end
```

>See `tol` in `isrotation`. 

My suggestion was not correct, we don't need the parameter `ϵ` as an argument.

Reply:
Ah, inside `if abs(sinhalftheta) < ϵ` block seems not depending on `t`? Isn't it a bug?

```julia
julia> q0 = UnitQuaternion(1., 0., 0., 0. )
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(1.0, 0.0, 0.0, 0.0):
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0

julia> q1 = UnitQuaternion(1., 0.0001,  0., 0. )
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(1.0, 0.0001, 0.0, 0.0):
 1.0  0.0      0.0
 0.0  1.0     -0.0002
 0.0  0.0002   1.0

julia> slerp(q0,q1,0) == slerp(q0,q1,1)
true
```

I think we can drop the if statement and don't need to define `ϵ`.

Reply:
>We can just alert the users of this in the docstring, and keep the slerp function simpler and faster. Thoughts?

I agree with that. More documentation is always great! Could you add some examples to [`docs/src/functions.md`](https://github.com/JuliaGeometry/Rotations.jl/blob/master/docs/src/functions.md)?